src/arch/riscv/insts/vector.cc

```diff
@@ -41,6 +41,74 @@ namespace gem5
 namespace RiscvISA
 {
 
+void roundUnsignedInteger(__uint128_t &result, uint32_t xrm, int gb) {
+  do {
+    const uint64_t lsb = 1UL << (gb);
+    const uint64_t lsb_half = lsb >> 1;
+    switch (xrm) {
+      case VectorRoundingMode::RoundToNearestUp:
+        result += lsb_half;
+        break;
+      case VectorRoundingMode::RoundToNearestEven:
+        if ((result & lsb_half) && ((result & (lsb_half - 1)) || (result & lsb))) {
+          result += lsb;
+        }
+        break;
+      case VectorRoundingMode::RoundDown:
+        break;
+      case VectorRoundingMode::RoundToOdd:
+        if (result & (lsb - 1)) {
+          result |= lsb;
+        }
+        break;
+      default:
+        printf("error: unknown vector rounding mode\n");
+        exit(1);
+    }
+  } while (0);
```

Comment:
We don't need `do while` here since it's not a macro.

Reply:
Fixed